gms/feature_service.hh

```diff
@@ -178,6 +179,22 @@ public:
         return bool(_uda);
     }
 
+    // Historically max_result_size contained only two fields: soft_limit and
+    // hard_limit. It was somehow obscure because for normal paged queries both
+    // fields were equal and meant page size. For unpaged queries and reversed
+    // queries soft_limit was used to warn when the size of the result exceeded
```

Comment:
So, why not say, when the entire cluster supports native reversed queries, reverse queries behave like normal queries?

Reply:
More generally the coordinator doesn't pick-and-choose what to send to the replica now, it always sends both the normal page size and the soft/hard limits for outliers. The replica then uses those that it needs.

Reply:
This has merit. I'd like to see the rules spelled out about how to derive page_size when it is not provided.

Reply:
I agree, the coordinator should not make assumptions about what the replica will need. It should supply all the parameters.